### PR TITLE
Solution [Svystun Bohdan]

### PR DESCRIPTION
**Pull request by BSvystun2707**

### CODE REVIEW

File: src/restoreNames.test.js [hunk 1]
```diff
@@ -1,11 +1,66 @@
...
+          lastName: 'Holy',
+          fullName: 'Jack Holy',
+        },
+        {
+          lastName: 'Adams',
+          fullName: 'Mike Adams',
+        },
+      ];
 
-  });
```
Comment (VeronikaSynieokaia) [inline src/restoreNames.test.js:8]: The 'restoreNames' function is a pure function and should not mutate its input. However, this test is set up to mutate the input array. Consider cloning the input array before passing it to the function, and then test if the returned array has the expected changes.

File: src/restoreNames.test.js [hunk 2]
```diff
@@ -1,11 +1,66 @@
...
+          fullName: 'John Doe',
+        },
+        {
+          firstName: 'Jane',
+          lastName: 'Smith',
+          fullName: 'Jane Smith',
+        },
+      ];
 
-  // write tests here
```
Comment (VeronikaSynieokaia) [inline src/restoreNames.test.js:10]: This test case only checks if firstName exists and doesn't mutate it. Please add test cases for different conditions such as firstName is null, firstName is an empty string, and firstName is a non-empty string but different from the one in fullName.

**Review by VeronikaSynieokaia — CHANGES_REQUESTED** [review]
check the comments please

File: src/restoreNames.test.js [hunk 3]
```diff
@@ -1,11 +1,104 @@
...
+      },
+      {
+        firstName: 'Mike',
+        lastName: 'Adams',
+        fullName: 'Mike Adams',
+      },
+    ]);
+  });
+
+  it('should restore firstName for users with undefined', () => {
```
Comment (VeronikaSynieokaia) [inline src/restoreNames.test.js:35]: The test case at line 37 has the same name as the test case at line 8. It's better to have unique names for each test case for better readability and understanding of what each test is doing.

**Review by VeronikaSynieokaia — APPROVED** [review]
well done
